test/plots/rectanglePlotTests.ts

```diff
@@ -2,444 +2,418 @@
+      it("retrieves the entities that intersect with the bounding box", () => {
+        plot.addDataset(new Plottable.Dataset(DATA));
+        plot.renderTo(svg);
 
-      yScale.domain(["A"]);
-      assert.deepEqual(xScale.domain(), [0, 1], "y domain includes only the visible segment (first)");
+        let entities = plot.entitiesIn({
+          topLeft: { x: xScale.scale(1.5), y: yScale.scale(2.5) },
+          bottomRight: { x: xScale.scale(2.5), y: yScale.scale(1.5) } });
+        assert.lengthOf(entities, 2, "retrieved 2 entities intersect with the box");
+        assert.strictEqual(entities[0].index, 1, "the entity of index 1 is retrieved");
+        assert.strictEqual(entities[1].index, 2, "the entity of index 2 is retrieved");
+      });
 
-      yScale.domain(["B"]);
-      assert.deepEqual(xScale.domain(), [1, 2], "y domain includes only the visible segment (second)");
+      it("retrieves the entities that intersect with the given ranges", () => {
+        plot.addDataset(new Plottable.Dataset(DATA));
+        plot.renderTo(svg);
 
-      svg.remove();
+        let entities = plot.entitiesIn(
+          {min: xScale.scale(1.5), max: xScale.scale(2.5)},
+          {min: yScale.scale(2.5), max: yScale.scale(1.5)});
+        assert.lengthOf(entities, 2, "retrieved 2 entities intersect with the box");
+        assert.strictEqual(entities[0].index, 1, "the entity of index 1 is retrieved");
+        assert.strictEqual(entities[1].index, 2, "the entity of index 2 is retrieved");
+      });
     });
 
-    it("autorangeMode(\"y\")", () => {
-      let staggeredData = [
-        { x: "A", y: 0, y2: 1 },
-        { x: "B", y: 1, y2: 2 }
-      ];
+    describe("RectanglePlot - autorangeMode()", () => {
+      let svg: d3.Selection<void>;
 
-      let xScale = new Plottable.Scales.Category();
-      let yScale = new Plottable.Scales.Linear();
-      yScale.padProportion(0);
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(300, 300);
+      });
 
-      let plot = new Plottable.Plots.Rectangle();
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.y2(function(d) { return d.y2; });
-      plot.addDataset(new Plottable.Dataset(staggeredData));
-      plot.autorangeMode("y");
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      plot.renderTo(svg);
+      afterEach(() => {
+        svg.remove();
+      });
 
-      assert.deepEqual(yScale.domain(), [0, 2], "y domain includes both visible segments");
+      it("autorangeMode(\"x\")", () => {
+        let staggeredData = [
+          { y: "A", x: 0, x2: 1 },
+          { y: "B", x: 1, x2: 2 }
+        ];
 
-      xScale.domain(["A"]);
-      assert.deepEqual(yScale.domain(), [0, 1], "y domain includes only the visible segment (first)");
+        let xScale = new Plottable.Scales.Linear();
+        let yScale = new Plottable.Scales.Category();
+        xScale.padProportion(0);
```

Comment:
To avoid computing the padding as well. 

You look at the data and say "oh yea, lowest value is 0, highest one is 2, so I expect the domain to be [0, 2] (instead of [-0.2, 2.2])" Make it as easier for the reader to figure out why is the answer of the assert the one it is.